foundry/foundry.py

```diff
@@ -365,12 +392,201 @@ def publish(self, foundry_metadata, data_source, title, authors, update=False,
+            logger.error(e.message)  # NOTE: known issue where user can still write to endpoint if this fails
+        return rule_id
+
+    def _https_upload(self, local_data_path, dest_path="/tmp", endpoint_id="82f1b5c6-6e9b-11e5-ba47-22000b92c6ec"):
+        """Upload a dataset via HTTPS to a Globus endpoint
+        Args:
+            local_data_path (str): The path to the local data to upload. Can be relative or absolute.
+            dest_path (str): The path to the destination folder on the Globus endpoint. Default is "/tmp".
+            endpoint_id (str): A UUID designating the exact Globus endpoint. Can be obtained via the Globus Web UI or
+                the SDK. Default is the NCSA UUID '82f1b5c6-6e9b-11e5-ba47-22000b92c6ec'.
+        Returns
+        -------
+            (str): Globus data source URL (ie the URL that points to the data on a Globus endpoint)
+        """
+        # get URL for Globus endpoint location
+        endpoint = self.transfer_client.get_endpoint(endpoint_id)  # gets info for NCSA endpoint
+        https_base_url = endpoint['https_server']
+
+        # Submit data (folders of files or an independent file) to be written to endpoint
+        if os.path.isdir(local_data_path):
+            self._upload_folder(local_data_path, https_base_url, dest_path, endpoint_id)
+        elif os.path.isfile(local_data_path):
+            self._upload_file(local_data_path, https_base_url, dest_path, endpoint_id)
+        else:
+            raise IOError(f"Data path '{local_data_path}' is of unknown type")
+
+        # return the data source URL for publication to MDF
+        return self.make_globus_link(endpoint_id, dest_path)
+
+    def _upload_folder(self, local_data_path, https_base_url, parent_dest_path, endpoint_id):
+        """Upload a folder to a Globus endpoint using HTTPS
+        Args:
+            local_data_path (str): The path to the local data to upload. Can be relative or absolute.
+            https_base_url (str): The URL for a given Globus endpoint.
+            parent_dest_path (str): The path to the parent folder to be written to on the given endpoint. The contents
+                of 'local_data_path' will be written here, including subdirectories.
+            endpoint_id (str): The UUID designating the exact Globus endpoint. Can be obtained via the Globus Web UI or
+                the SDK. This must be the same endpoint pointed to by the https_base_url.
+        Returns
+        -------
+            (list): A list of all of the HTTPS PUT request results (dicts) from the uploads
+        """
+        results = []
+        # initialize destination path as the parent destination path
+        dest_path = parent_dest_path
+
+        # walk through each child directory in the designated local data folder
+        for root, directories, files in os.walk(local_data_path):
+            # update destination path if we have walked into a child directory
+            if root is not local_data_path:
```

Comment:
If these are strings, it should be "!=" and not "is not." (you have have two different strings objects with the same value)

Reply:
`os.walk` doesn't guarantee it will pass the same object back as input.

Reply:
whoops good catch, thank you Logan!